pkg/systemconfig/types.go

```diff
@@ -49,30 +50,39 @@ var (
 	registriesCertsDir,
 	registriesConfPath,
 	policyConfPath string
+	rwMutex sync.RWMutex
 )
 
 func DockerCertsDir() string {
+	rwMutex.RLock()
+	defer rwMutex.RUnlock()
 	if dockerCertsDir == "" {
 		dockerCertsDir = lookupEnvOr("DOCKER_CERTS_DIR", "/etc/docker/certs.d")
 	}
 	return dockerCertsDir
 }
 
 func RegistryCertsDir() string {
+	rwMutex.Lock()
+	defer rwMutex.Unlock()
 	if registriesCertsDir == "" {
 		registriesCertsDir = lookupEnvOr("REGISTRIES_CERTS_DIR", "/etc/containers/registries.d")
 	}
 	return registriesCertsDir
 }
 
 func RegistriesConfPath() string {
+	rwMutex.RLock()
+	defer rwMutex.RUnlock()
 	if registriesConfPath == "" {
 		registriesConfPath = lookupEnvOr("REGISTRIES_CONF_PATH", "/etc/containers/registries.conf")
 	}
 	return registriesConfPath
 }
 
 func PolicyConfPath() string {
+	rwMutex.RLock()
+	defer rwMutex.RUnlock()
 	if policyConfPath == "" {
 		policyConfPath = lookupEnvOr("POLICY_CONF_PATH", "/etc/containers/policy.json")
 	}
```

Comment:
```suggestion
	rwMutex.RLock()
	if policyConfPath != "" {
		defer rwMutex.RUnlock()
	    return policyConfPath
	}
	rwMutex.RUnlock()
	rwMutex.Lock()
	defer rwMutex.Unlock()
	if policyConfPath == "" { // avoid race condition in-between rwMutex.RUnlock and rwMutex.Lock
	  policyConfPath = lookupEnvOr("POLICY_CONF_PATH", "/etc/containers/policy.json")
	}
```

We should write lock when writing.